homeassistant/components/velux/__init__.py

```diff
@@ -1,54 +1,35 @@
 """Support for VELUX KLF 200 devices."""
-import logging
-
 from pyvlx import PyVLX, PyVLXException
-import voluptuous as vol
-
-from homeassistant.const import (
-    CONF_HOST,
-    CONF_PASSWORD,
-    EVENT_HOMEASSISTANT_STOP,
-    Platform,
-)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, EVENT_HOMEASSISTANT_STOP
 from homeassistant.core import HomeAssistant, ServiceCall, callback
-from homeassistant.helpers import discovery
-import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.entity import Entity
-from homeassistant.helpers.typing import ConfigType
 
-DOMAIN = "velux"
-DATA_VELUX = "data_velux"
-PLATFORMS = [Platform.COVER, Platform.LIGHT, Platform.SCENE]
-_LOGGER = logging.getLogger(__name__)
-
-CONFIG_SCHEMA = vol.Schema(
-    {
-        DOMAIN: vol.Schema(
-            {vol.Required(CONF_HOST): cv.string, vol.Required(CONF_PASSWORD): cv.string}
-        )
-    },
-    extra=vol.ALLOW_EXTRA,
-)
```

Comment:
You need to keep the existing schema with a deprecating period.
If the user has a YAML entry, it should raise an issue in the issue-registry and automatically import into a config-entry.
See #85797

Reply:
Ok. Just one more question about the process, if the YAML configuration for example expires in version 2023.7.0. Do I create the PR that removes the obsolete code now as soon as this PR is done and you merge it at the right time or do I have to create the PR in June to get the right timing?